src/Compiler/LatteToPhpCompiler.php

```diff
@@ -4,24 +4,30 @@
 
 namespace Efabrica\PHPStanLatte\Compiler;
 
+use Composer\InstalledVersions;
 use Efabrica\PHPStanLatte\Compiler\Compiler\CompilerInterface;
 use Efabrica\PHPStanLatte\Template\Template;
 use InvalidArgumentException;
+use Latte\Engine;
 
 final class LatteToPhpCompiler
 {
     private string $tmpDir;
 
+    private string $cacheKey;
+
     private CompilerInterface $compiler;
 
     private Postprocessor $postprocessor;
 
     public function __construct(
         ?string $tmpDir,
+        string $cacheKey,
         CompilerInterface $compiler,
         Postprocessor $postprocessor
     ) {
         $this->tmpDir = $tmpDir ?? sys_get_temp_dir() . '/phpstan-latte';
+        $this->cacheKey = $cacheKey;
```

Comment:
I would add it here = $cacheKey . (class_exists(InstalledVersions::class) ? json_encode(InstalledVersions::getAllRawData()) : '');